src/CalcViewModel/UnitConverterViewModel.cpp

```diff
@@ -990,3 +993,56 @@ String ^ SupplementaryResult::GetLocalizedAutomationName()
     auto format = AppResourceProvider::GetInstance()->GetResourceString("SupplementaryUnit_AutomationName");
     return LocalizationStringUtil::GetLocalizedString(format, this->Value, this->Unit->Name);
 }
+
+Unit^ UnitConverterViewModel::Unit1::get()
+{
+    return m_Unit1;
+}
+
+void UnitConverterViewModel::Unit1::set(Unit^ value)
+{
+    if (m_Unit1 == value)
+    {
+        return;
+    }
+
+    if (value == m_Unit2 && value != nullptr && m_Unit1 != nullptr)
+    {
+        auto formerValue = m_Unit1;
+        m_Unit1 = value;
+        //Be sure m_Unit1 is set before setting Unit2
```

Comment:
Why is this the case? An extra comment line giving more context might be helpful.